web/src/main/java/org/perfrepo/web/adapter/TestAdapter.java

```diff
@@ -0,0 +1,161 @@
+     *
+     * @return List of {@link TestDto} tests.
+     * @throws org.perfrepo.web.adapter.exceptions.AdapterException If anything bad happened.
+     */
+    SearchResult searchTests(TestSearchParams searchParams);
+
+    /**
+     * Add a metric to existing test. If a metric does not exist, it will be created.
+     *
+     * @param metric Metric {@link org.perfrepo.dto.metric.MetricDto} that will be added to the test.
+     * @param testId Test {@link TestDto} identifier.
+     * @return Metric {@link MetricDto} that was added to test.
+     * @throws org.perfrepo.web.adapter.exceptions.NotFoundException If the test does not exist.
+     * @throws org.perfrepo.web.adapter.exceptions.AdapterException If anything bad happened.
+     */
+    MetricDto addMetricToTest(MetricDto metric, Long testId);
+
+    /**
+     * Remove a metric from existing test.
+     *
+     * @param metricId Identifier of {@link org.perfrepo.dto.metric.MetricDto} metric that will be removed.
+     * @param testId Test {@link TestDto} identifier.
+     * @throws org.perfrepo.web.adapter.exceptions.NotFoundException If the test does not exist.
+     * @throws org.perfrepo.web.adapter.exceptions.AdapterException If anything bad happened.
+     */
+    void removeMetricFromTest(Long metricId, Long testId);
+
+    /**
+     * Subscribe the user to the test alerts.
+     *
+     * @param testId Test {@link TestDto} identifier.
+     * @param userId User {@link UserDto} identifier.
+     * @throws org.perfrepo.web.adapter.exceptions.NotFoundException If the test or user does not exist.
+     * @throws org.perfrepo.web.adapter.exceptions.AdapterException If anything bad happened.
+     */
+    void subscribeToTestAlerts(Long testId, Long userId);
+
+    /**
+     * Unsubscribe the user from the test alerts.
+     *
+     * @param testId Test {@link TestDto} identifier.
+     * @param userId User {@link UserDto} identifier.
+     * @throws org.perfrepo.web.adapter.exceptions.NotFoundException If the test or user does not exist.
+     * @throws org.perfrepo.web.adapter.exceptions.AdapterException If anything bad happened.
+     */
+    void unsubscribeFromTestAlerts(Long testId, Long userId);
+
+    /**
+     * Create new {@link AlertDto} alert for the test.
+     *
+     * @param testId Test {@link TestDto} identifier.
+     * @param alert
+     * @return
+     */
+    AlertDto createTestAlert(Long testId, AlertDto alert);
+
+    /**
+     * Update the {@link AlertDto} alert of the test.
+     *
+     * @param testId Test {@link TestDto} identifier.
+     * @param alert Alert {@link AlertDto} parameters.
+     * @return
+     */
+    AlertDto updateTestAlert(Long testId, AlertDto alert);
+
+    /**
+     * Delete {@link AlertDto} alert from the test.
+     *
+     * @param testId Test {@link TestDto} identifier.
+     * @param alertId Alert {@link AlertDto} identifier.
+     */
+    void deleteTestAlert(Long testId, Long alertId);
```

Comment:
Let's stick to the naming convention you used anywhere else - use remove* instead of delete*